plugin/kubernetes/controller.go

```diff
@@ -491,12 +432,16 @@ func (dns *dnsControl) detectChanges(oldObj, newObj interface{}) {
 		dns.updateModifed()
 	case *object.Pod:
 		dns.updateModifed()
+	case *object.Endpoints:
+		if !endpointsEquivalent(oldObj.(*object.Endpoints), newObj.(*object.Endpoints)) {
+			dns.updateModifed()
+		}
 	default:
 		log.Warningf("Updates for %T not supported.", ob)
 	}
 }
 
-func (dns *dnsControl) getServices(endpoints *object.Endpoints) []*object.Service {
+func (dns *dnsControl) getServices(endpoints *api.Endpoints) []*object.Service {
```

Comment:
why is this now an api.Endpoint instead of object?

Reply:
`GetServices` can construct the index key from either type. Changing it to `api.Endpoint ` allows us to pass a single argument to `recordDNSProgrammingLatency()` instead of two (both `api.Endpoint` and `object.Endpoint` for the same record).